src/lib.rs

```diff
@@ -284,6 +290,23 @@ pub fn main() -> Result<(), Error> {
                     knn = Some(n - 1);
                 }
             }
+            // Read in completeness file
+            let completeness_vec: Option<Vec<f64>> = if let Some(file_path) = completeness_file {
+            let contents = fs::read_to_string(file_path)?;
+            let mut completeness_map = HashMap::new();
+            
+            for line in contents.lines() {
+                if let Some((genome_id, completeness_str)) = line.split_once('\t') {
+                    if let Ok(completeness) = completeness_str.parse::<f64>() {
+                        completeness_map.insert(genome_id.to_string(), completeness);
+                    }
+                }
+            }
+                Some(create_completeness_vector(&completeness_map, &references))
+            } else {
+                None
+            };
```

Comment:
This should go in the io.rs file. Also see examples there for using `ReadBuf` -- you don't want to read the whole thing in as a string, it's better to stream it